src/ApiCore/Testing/GeneratedTest.php

```diff
@@ -35,10 +35,66 @@
 use Google\Protobuf\DescriptorPool;
 use Google\Protobuf\Internal\Message;
 use Google\Protobuf\Internal\RepeatedField;
+use PHPUnit\Framework\Exception;
 use PHPUnit\Framework\TestCase;
 
 abstract class GeneratedTest extends TestCase
 {
+    // Workaround for https://github.com/google/protobuf/issues/4761
+    public static function assertSame($expected, $actual, $message = '')
+    {
+        if ($expected instanceof \Google\Protobuf\Internal\Message) {
+            if (!$actual instanceof \Google\Protobuf\Internal\Message) {
+                throw new Exception("not \\Google\\Protobuf\\Internal\\Message object");
+                return;
```

Comment:
We shouldn't need to return here when an exception is thrown (applies to a few other instances below as well)